yt/yt/library/decimal/decimal.cpp

```diff
@@ -26,10 +144,138 @@ struct TDecimalTraits
     static constexpr T Nan = std::numeric_limits<T>::max();
     static constexpr T PlusInf = std::numeric_limits<T>::max() - 1;
     static constexpr T MinusInf = -PlusInf;
+};
+
+template <>
+struct TDecimalTraits<i256>
+{
+    static_assert(ValidDecimalUnderlyingInteger<i256>);
 
-    static constexpr T MinSpecialValue = PlusInf;
+    static constexpr i32 Int32Max = std::numeric_limits<i32>::max();
+    static constexpr ui32 UInt32Max = std::numeric_limits<ui32>::max();
+    static constexpr i256 Nan = {UInt32Max, UInt32Max, UInt32Max, UInt32Max, UInt32Max, UInt32Max, UInt32Max, Int32Max};
+    static constexpr i256 PlusInf = {UInt32Max - 1, UInt32Max, UInt32Max, UInt32Max, UInt32Max, UInt32Max, UInt32Max, Int32Max};
+    static constexpr i256 MinusInf = -PlusInf;
 };
 
+template <typename T>
+Y_FORCE_INLINE constexpr bool IsNegativeInteger(T value)
+{
+    static_assert(ValidDecimalUnderlyingInteger<T>);
+
+    if constexpr (std::is_same_v<T, i256>) {
+        return value.Parts[7] & (1u << 31);
+    } else {
+        return value < 0;
+    }
+}
+
+template <typename T>
+Y_FORCE_INLINE constexpr auto DecimalIntegerToUnsigned(T value)
```

Comment:
This function does not work for `ui256` and `ui128` and function below does not work for long signed types. Is it fine?

Reply:
Yep. There is even a static_assert.